core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -172,16 +174,54 @@ public void removeBlock(long blockId) {
   }
 
   /**
-   * @return generated {@link WorkerInfo} for this worker
+   * Gets the selected field information for this worker.
+   *
+   * @param fieldRange the client selected fields
+   * @param isLiveWorker the worker is live or not
+   * @return generated worker information
    */
-  public WorkerInfo generateClientWorkerInfo() {
-    return new WorkerInfo()
-        .setId(mId)
-        .setAddress(mWorkerAddress)
-        .setLastContactSec(
-            (int) ((CommonUtils.getCurrentMs() - mLastUpdatedTimeMs) / Constants.SECOND_MS))
-        .setState("In Service").setCapacityBytes(mCapacityBytes).setUsedBytes(mUsedBytes)
-        .setStartTimeMs(mStartTimeMs);
+  public WorkerInfo generateWorkerInfo(Set<WorkerInfoField> fieldRange, boolean isLiveWorker) {
+    WorkerInfo info = new WorkerInfo();
+    for (WorkerInfoField field : fieldRange != null ? fieldRange
```

Comment:
have a null check on `fieldRange` before for-loop rather than having conditional check here.

Reply:
Changed to 
```java
  Set<WorkerInfoField> checkedFieldRange = fieldRange != null ? fieldRange :
        new HashSet<>(Arrays.asList(WorkerInfoField.values()));
    for (WorkerInfoField field : checkedFieldRange) {
```
Thanks